src/registrar/management/commands/send_domain_invitations.py

```diff
@@ -0,0 +1,148 @@
+        else:
+            logger.debug("not sending emails")
+
+    def build_emails_to_send_array(self):
+        """this method sends emails to distinct usernames"""
+
+        # data structure to hold email context for a single email;
+        # transition_domains ordered by username, a single email_context
+        # may include information from more than one transition_domain
+        email_context = {"email": ""}
+
+        # loop through all transition_domains; group them by username
+        # into emails_to_send_array
+        for transition_domain in self.transition_domains:
+            # attempt to get the domain from domain objects; if there is
+            # an error getting the domain, skip this domain and add it to
+            # domains_with_errors
+            try:
+                domain = Domain.objects.get(name=transition_domain.domain_name)
+                # if prior username does not match current username
+                if (
+                    not email_context["email"]
+                    or email_context["email"] != transition_domain.username
+                ):
+                    # if not first in list of transition_domains
+                    if email_context["email"]:
+                        # append the email context to the emails_to_send array
+                        self.emails_to_send.append(copy.deepcopy(email_context))
+                    email_context["domains"] = []
+                email_context["email"] = transition_domain.username
+                email_context["domains"].append(
+                    {
+                        "name": transition_domain.domain_name,
+                        "url": settings.BASE_URL
+                        + reverse("domain", kwargs={"pk": domain.id}),
+                    }
+                )
+            except Exception as err:
+                # error condition if domain not in database
+                self.domains_with_errors.append(
+                    copy.deepcopy(transition_domain.domain_name)
+                )
+                logger.error(
+                    f"error retrieving domain {transition_domain.domain_name}: {err}"
+                )
+        # if there are at least one more transition domains than errors,
+        # then send one more
+        if len(self.transition_domains) > len(self.domains_with_errors):
+            self.emails_to_send.append(email_context)
+
+    def send_emails(self):
+        for email_data in self.emails_to_send:
+            self.send_email(email_data)
+
+    def send_email(self, email_data):
+        try:
+            send_templated_email(
+                "emails/transition_domain_invitation.txt",
+                "emails/transition_domain_invitation_subject.txt",
+                to_address=email_data["email"],
+                context={
+                    "domains": email_data["domains"],
+                },
+            )
+            # if log level set to debug, success message is logged
```

Comment:
Ah, I see why you may be doing logger.debug in your handle function. Will this script always be restricted to debug only, or would you like to only restrict information but allow it to run otherwise?

Reply:
rethinking this.  i used debug intentionally so that this script could be run without the noise, and only errors would print.  but since the successful information is important to log for accountability, i have modified all of the debugs to infos.